_LowLevelCode/cpp/cpp_communicator/serialiser/c_deserialise.cpp

```diff
@@ -0,0 +1,363 @@
+/*=========================================================
+ * c_deserialise.cpp
+ * Deserialise serialised data back into a MATLAB object
+ *
+ * See also:
+ * hlp_serialize
+ * hlp_deserialize
+ *
+ * This is a MEX-file for MATLAB.
+ *=======================================================*/
+
+#include <iostream>
+#include <cstring>
+#include <cmath>
+#include <vector>
+#include "cpp_serialise.hpp"
+
+size_t memPtr;
+
+template<typename T>
+inline void deser(const uint8_t* data, std::vector<T> output, const double amount) {
+  memcpy(output.data(), &data[memPtr], amount);
+  memPtr += amount;
+}
+
+inline void deser(const uint8_t* data, void* output, const double amount) {
+  memcpy(output, &data[memPtr], amount);
+  memPtr += amount;
+}
+
+inline void read_data(uint8_t* data, mxArray* output, const size_t elemSize, const size_t nElem) {
+  if (mxIsComplex(output)) {
+    // Size of a complex component is half that of the whole double
```

Comment:
Precisely, the size of a complex in the read is 2 doubles, hence the size of a complex component (be that real or imag) is half of that. I will comment to make that more clear.

Reply:
That sounds clear enough. Just pointing out that your comment says "half that of the whole double" - which I would think was a float. If you had said "half of the 2 doubles" it would be clearer.